vars/withPipeline.groovy

```diff
@@ -119,6 +119,14 @@ def call(type, String product, String component, Closure body) {
             )
           }
 
+          highLevelDataSetup(
+            appPipelineConfig: pipelineConfig,
+            pipelineCallbacksRunner: callbacksRunner,
+            builder: pipelineType.builder,
+            stage: DockerImage.DeploymentStage.PREVIEW,
```

Comment:
I don't think this is quite right,

aren't you after environment?

https://github.com/hmcts/cnp-jenkins-library/blob/CCD-1131/vars/withPipeline.groovy#L154

https://github.com/hmcts/cnp-jenkins-library/blob/CCD-1131/vars/withPipeline.groovy#L205

Reply:
We call highLevelDataSetup 2 times onMaster; one is for AAT and one is for PROD. When I checked similar stages, I found sectionPromoteBuildToStage() and stage type is passed in the same which I used here.

Reply:
that's tied to docker image code though, I can't see why you would use that here, all you are using this for is the environment name isn't it?

Reply:
Yes, it is and I though to pick the name from an enum would be more convenient. But, I will create a new enum in highLevelDataSetup and will pick the names from there.

Reply:
you shouldn't need to do that?

you have nonProdName, and prodName from `environment` and the return value from `onAutoDeployBranch`

Reply:
Could you please confirm these for me?

- onPR: 
        environment.nonProdName -> preview

- onMaster:
       environment.nonProdName -> aat
       environment.prodName -> prod

- onAutoDeployBranch
      environment.nonProdName -> depends on the environment

Reply:
values are here:
https://github.com/hmcts/cnp-jenkins-library/blob/master/src/uk/gov/hmcts/contino/Environment.groovy#L24

Reply:
resolved in [b7ebe7a5e40a9e22fc3300f30a00582f0183cfb1](https://github.com/hmcts/cnp-jenkins-library/pull/747/commits/b7ebe7a5e40a9e22fc3300f30a00582f0183cfb1)